src/com/palmergames/bukkit/towny/db/TownySQLSource.java

```diff
@@ -77,41 +80,46 @@ public TownySQLSource(Towny plugin, TownyUniverse universe, String type) {
 		/*
 		 * Setup SQL connection
 		 */
-		String hostname = TownySettings.getSQLHostName();
-		String port = TownySettings.getSQLPort();
-		String flags = TownySettings.getSQLFlags();
 		db_name = TownySettings.getSQLDBName();
 		tb_prefix = TownySettings.getSQLTablePrefix().toUpperCase();
 		
-		String driver1;
-		if (this.type.equals("h2")) {
-
-			driver1 = "org.h2.Driver";
-			this.dsn = ("jdbc:h2:" + dataFolderPath + File.separator + db_name + ".h2db;AUTO_RECONNECT=TRUE");
-			username = "sa";
-			password = "sa";
-
-		} else if (this.type.equals("mysql")) {
-
-			driver1 = "com.mysql.jdbc.Driver";
-			this.dsn = ("jdbc:mysql://" + hostname + ":" + port + "/" + db_name + flags);
-			username = TownySettings.getSQLUsername();
-			password = TownySettings.getSQLPassword();
-
-		} else {
-
-			driver1 = "org.sqlite.JDBC";
-			this.dsn = ("jdbc:sqlite:" + dataFolderPath + File.separator + db_name + ".sqldb");
-			username = "";
-			password = "";
-
-		}
+		this.dsn = ("jdbc:mysql://" + TownySettings.getSQLHostName() + ":" + TownySettings.getSQLPort() + "/" + db_name + TownySettings.getSQLFlags());
+		this.config = new HikariConfig();
+		
+		config.setPoolName("Towny MySQL");
+		config.setJdbcUrl(this.dsn);
+
+		username = TownySettings.getSQLUsername();
+		password = TownySettings.getSQLPassword();
+
+		config.setUsername(username);
+		config.setPassword(password);
+
+		config.addDataSourceProperty("cachePrepStmts", "true");
+		config.addDataSourceProperty("prepStmtCacheSize", "250");
+		config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		config.addDataSourceProperty("useServerPrepStmts", "true");
+		config.addDataSourceProperty("useLocalSessionState", "true");
+		config.addDataSourceProperty("rewriteBatchedStatements", "true");
+		config.addDataSourceProperty("cacheResultSetMetadata", "true");
+		config.addDataSourceProperty("cacheServerConfiguration", "true");
+		config.addDataSourceProperty("elideSetAutoCommits", "true");
+		config.addDataSourceProperty("maintainTimeStats", "false");
+		config.addDataSourceProperty("alwaysSendSetIsolation", "false");
```

Comment:
It would appear this line doesn't do anything: 
```
Note that useLocalSessionState=true will force the same behavior as alwaysSendSetIsolation=false, regardless of how alwaysSendSetIsolation is set.
```

Reply:
Solved this by removing it since not needed.